packages/google_sign_in/android/src/main/java/io/flutter/plugins/googlesignin/GoogleSignInPlugin.java

```diff
@@ -342,7 +346,26 @@ public void run(Future<String> tokenFuture) {
                 // instead of the value we cached during sign in. At least, that's
                 // how it works on iOS.
                 result.success(tokenResult);
-              } catch (ExecutionException e) {
+              } catch (final ExecutionException e) {
+                if (shouldRecoverAuth && e.getCause() instanceof UserRecoverableAuthException) {
```

Comment:
If there's a UserRecoverableAuthException but shouldRecoverAuth is false, it might still be useful to report the exception to the Dart side (without the recover method that you had before).